src/acl.c

```diff
@@ -652,6 +652,7 @@ sds ACLDescribeUser(user *u) {
     if (u->flags & USER_FLAG_ALLCHANNELS) {
         res = sdscatlen(res,"&* ",3);
     } else {
+        res = sdscatlen(res,"nochannels ",11);
```

Comment:
```suggestion
        if (server.acl_pubsub_default & USER_FLAG_ALLCHANNELS)
            res = sdscatlen(res,"resetchannels ",14);
```

I think the real problem is when the default `acl-pubsub-default resetchannels`:

```
127.0.0.1:6382> acl setuser hello resetchannels
OK
127.0.0.1:6382> acl list
1) "user default on nopass sanitize-payload ~* &* +@all"
2) "user hello off -@all"
```

We lost `resetchannels`.

Reply:
I guess I wasn't clear enough in the above description. `resetchannels` is only used to revert the user to the default state. In the code, the user has access to only selected channels hence, `nochannels` is prefixed first and then followed up with each channel name. This `nochannels` is required to maintain the idempotency upon `ACL SAVE/ACL LOAD`.

when the flag `acl-pubsub-default resetchannels` or `acl-pubsub-default nochannels` 
The output will be the following.

```
127.0.0.1:6379> acl setuser hello resetchannels
OK
127.0.0.1:6379> acl list
1) "user default on nopass ~* &* +@all"
2) "user hello off nochannels -@all"
```